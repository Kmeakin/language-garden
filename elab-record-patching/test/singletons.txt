let intro-sing := (fun A x := x) : fun (A : Type) (x : A) -> A [= x ];
let elim-sing := (fun A x sing-x := sing-x) : fun (A : Type) (x : A) (sing-x : A [= x ]) -> A;

let sing-tm-1 :=
  (fun A P f prf := prf) :
    fun (A : Type)
        (P : (fun (x : A) -> A [= x ]) -> Type)
        (f : fun (x : A) -> A [= x ])
        (prf : P (fun x := x))
    -> P f;

Type
