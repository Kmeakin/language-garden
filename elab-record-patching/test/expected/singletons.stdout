<input> : Type :=
  let intro-sing : fun (A : Type) (x : A) -> A [= x] :=
    fun A x := #sing-intro x;
  let elim-sing : fun (A : Type) (x : A) (sing-x : A [= x]) -> A :=
    fun A x sing-x := #sing-elim sing-x x;
  let sing-tm-1 :
        fun (A : Type) (P : (fun (x : A) -> A [= x]) -> Type)
            (f : fun (x : A) -> A [= x]) (prf : P (fun x := #sing-intro x))
            -> P f
  := fun A P f prf := prf; Type
